qsiprep/tests/test_utils_grouping.py

```diff
@@ -117,22 +188,402 @@ def test_get_entity_groups_without_multipartid(tmpdir):
+            ],
+            'dwi_series_pedir': 'j-',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-PA_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+        {
+            'concatenated_bids_name': 'sub-01_dir-PA',
+            'dwi_series': [
+                'sub-01_dir-PA_dwi.nii.gz',
+            ],
+            'dwi_series_pedir': 'j',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-AP_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+    ]
+    check_expected(scan_groups, expected)
+
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=True,
+        combine_scans=False,
+        ignore_fieldmaps=False,
+        concatenate_distortion_groups=False,
+    )
+    expected = [
+        {
+            'concatenated_bids_name': 'sub-01',
+            'dwi_series': [
+                'sub-01_dir-PA_dwi.nii.gz',
+            ],
+            'dwi_series_pedir': 'j',
+            'fieldmap_info': {
+                'epi': [
+                    'sub-01_dir-AP_epi.nii.gz',
+                    'sub-01_dir-PA_epi.nii.gz',
+                ],
+                'rpe_series': [
+                    'sub-01_dir-AP_run-1_dwi.nii.gz',
+                    'sub-01_dir-AP_run-2_dwi.nii.gz',
+                ],
+                'suffix': 'rpe_series',
+            },
+        },
+    ]
+    check_expected(scan_groups, expected)
+
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=False,
+        combine_scans=False,
+        ignore_fieldmaps=False,
+        concatenate_distortion_groups=False,
+    )
+    expected = [
+        {
+            'concatenated_bids_name': 'sub-01_dir-AP_run-1',
+            'dwi_series': ['sub-01_dir-AP_run-1_dwi.nii.gz'],
+            'dwi_series_pedir': 'j-',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-PA_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+        {
+            'concatenated_bids_name': 'sub-01_dir-AP_run-2',
+            'dwi_series': ['sub-01_dir-AP_run-2_dwi.nii.gz'],
+            'dwi_series_pedir': 'j-',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-PA_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+        {
+            'concatenated_bids_name': 'sub-01_dir-PA',
+            'dwi_series': ['sub-01_dir-PA_dwi.nii.gz'],
+            'dwi_series_pedir': 'j',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-AP_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+    ]
+    check_expected(scan_groups, expected)
+
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=True,
+        combine_scans=True,
+        ignore_fieldmaps=True,
```

Comment:
This means ignore distortion correction, not just ignore field maps. Should rename, but logic makes sense.

Reply:
Opened #965.